compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2182,6 +2188,21 @@ public void visit(BLangFunction funcNode) {
         }
     }
 
+    private SymbolEnv fieldsRemovedEnv(SymbolEnv currentEnv) {
+        Scope currentScope = currentEnv.scope;
```

Comment:
We can probably avoid creating a new env if the class/object doesn't have fields.

Reply:
Changed